src/daemon/rpmostreed-os.c

```diff
@@ -77,6 +85,147 @@ sysroot_changed (RpmostreedSysroot *sysroot,
+    {
+      g_ptr_array_add (actions, "org.projectatomic.rpmostree1.install-uninstall-packages");
+    }
+  else if (g_strcmp0 (method_name, "UpdateDeployment") == 0)
+    {
+      g_autoptr(GVariant) modifiers = g_variant_get_child_value (parameters, 0);
+      g_autoptr(GVariant) options = g_variant_get_child_value (parameters, 1);
+      g_auto(GVariantDict) modifiers_dict;
+      g_auto(GVariantDict) options_dict;
+      g_variant_dict_init (&modifiers_dict, modifiers);
+      g_variant_dict_init (&options_dict, options);
+      const char *refspec =
+        vardict_lookup_ptr (&modifiers_dict, "set-refspec", "&s");
+      const char *revision =
+        vardict_lookup_ptr (&modifiers_dict, "set-revision", "&s");
+      g_autofree char **install_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "install-packages", "^a&s");
+      g_autofree char **uninstall_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "uninstall-packages", "^a&s");
+      g_autofree const char *const *override_replace_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "override-replace-packages", "^a&s");
+      g_autofree const char *const *override_remove_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "override-remove-packages", "^a&s");
+      g_autofree const char *const *override_reset_pkgs =
+        vardict_lookup_ptr (&modifiers_dict, "override-reset-packages", "^a&s");
+      g_autoptr(GVariant) install_local_pkgs =
+        g_variant_dict_lookup_value (&modifiers_dict, "install-local-packages",
+                                     G_VARIANT_TYPE("ah"));
+      g_autoptr(GVariant) override_replace_local_pkgs =
+        g_variant_dict_lookup_value (&modifiers_dict, "override-replace-local-packages",
+                                     G_VARIANT_TYPE("ah"));
+      gboolean no_pull_base =
+        vardict_lookup_bool (&options_dict, "no-pull-base", FALSE);
+      gboolean no_overrides =
+        vardict_lookup_bool (&options_dict, "no-overrides", FALSE);
+
+      if (refspec != NULL)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.rebase");
+      else if (revision != NULL)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.deploy");
+      else if (!no_pull_base)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.upgrade");
+
+      if (install_pkgs != NULL || uninstall_pkgs != NULL)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.install-uninstall-packages");
+
+      if (install_local_pkgs != NULL && g_variant_n_children (install_local_pkgs) > 0)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.install-local-packages");
+
+      if (override_replace_pkgs != NULL || override_remove_pkgs != NULL || override_reset_pkgs != NULL ||
+          (override_replace_local_pkgs != NULL && g_variant_n_children (override_replace_local_pkgs) > 0) ||
+          no_overrides)
+        g_ptr_array_add (actions, "org.projectatomic.rpmostree1.override");
+    }
```

Comment:
Sure, easy enough to add, although we already initialize it to `FALSE` in the `gboolean authorized = FALSE;` line above, which makes having the `else` block unnecessary.